public/js/console-game.js

```diff
@@ -1,76 +1,124 @@
-function Game(name, options) {
-
-  var actions = options.actions;
-  var steps = options.steps;
-  var count = 0;
-  var correctAnswer;
-  var causeOfDeath;
-
-  var utils = {
-    outputChoices: function (arr) {
-      var output = '';
-      var length = arr.length - 1;
-      arr.forEach(function (e, i) {
-        output += (i !== length) ? e + ', ' : e;
-      });
-      return output;
-    },
-    readInput: function (input) {
-      console.log(input);
-      if (input !== correctAnswer) {
-        console.log(causeOfDeath);
-      } else {
-        play(count);
-      }
-    },
-    showPrompt: function (option) {
-      console.log(option.prompt + '(' + utils.outputChoices(option.options) + ')');
-    },
-    setCorrectAnswer: function (step) {
-      if (step.answer !== undefined) {
-        var answer = actions[step.prompt].options[step.answer];
-        correctAnswer = answer;
-      }
-    },
-    setCauseOfDeath: function (step) {
-      if (step.die) {
-        var die = step.die;
-        causeOfDeath = die;
-      }
-    },
-    outputMessage: function (step) {
-      console.log(step.message);
-      if (step.prompt) {
-        utils.showPrompt(actions[step.prompt]);
-      }
-    }
-  };
-
-  var play = function () {
-    var step = steps[count];
-    utils.setCorrectAnswer(step);
-    utils.setCauseOfDeath(step);
-    utils.outputMessage(step);
-    count++;
-  };
+function NewGame(name, options)
+{
+	var game = new Game(name, options);
```

Comment:
Guess we'll have to define 2 space or 4 space for this project... I had been doing 2 spaces.